storage/cloud-client/bucket_lock.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+from google.cloud import storage
+
+
+def set_retention_policy(bucket_name, retention_period):
```

Comment:
Add commented lines (inside the region tags)  showing how out of scope variables are instantiated:

```python
# bucket_name = "my-bucket-name"
# retention_period  = storage.option.whatever
```

(same for other functions)